classes/ActionScheduler_WPCLI_QueueRunner.php

```diff
@@ -207,4 +213,16 @@ protected function stop_the_insanity( $sleep_time = 0 ) {
 			call_user_func( array( $wp_object_cache, '__remoteset' ) ); // important
 		}
 	}
+
+	/**
+	 * Get timestamp if property 'print_timestamp' is truthy.
+	 *
+	 * @uses ActionScheduler_DateTime::getCliTimestamp()
+	 * @return null|string
+	 */
+	protected function maybe_get_timestamp() {
```

Comment:
Similar to something @rrennick mentioned - prepending the date/time could be done in a separate function like this one to reduce the duplicate code prepending above (and make the approach consistent, e.g. always use placeholders of the prepend operator `.`).

`maybe_get_timestamp()` could be repurposed to something like `maybe_prepend_date( $message )` that both checks `$this->print_timestamp` and then prepends `as_get_datetime_object()->getCliTimestamp()` to the `$message` if needed.

Or ideally, this could be moved into a separate function outside the class like `as_maybe_prepend_date( $message, $print_date )` so that it can also be used by `ActionScheduler_WPCLI_Scheduler_command`.

Reply:
Yeah, I would've preferred to have put this in a class that was shared between the runner and the scheduler, but because they didn't share anything, I wasn't sure what was preferred.

I've implemented `as_get_formatted_cli_timestamp()` in 42a9506b83cc5043497a52f9663ebdecc50bed79.